### PR TITLE
Support storing YAML job templates in the database

### PR DESCRIPTION
**Pull request by kalikiana**
- Add `template` column to `JobGroups` schema (and migration scripts).
- Support `reference` parameter to `update` route which now aborts if the current template doesn't match the reference document, so as to allow handling of conflicts.
- Update schema to remove top-level group name - as a consequence groups can't be created through the route
- `YAMLRenderer` now takes a string instead of a hash

Fixes: [poo#46673](https://progress.opensuse.org/issues/46673)

### CODE REVIEW

File: lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm [hunk 1]
```diff
@@ -236,22 +266,29 @@ sub update {
     my $machines              = $schema->resultset('Machines');
     my $test_suites           = $schema->resultset('TestSuites');
     my $products              = $schema->resultset('Products');
-    foreach my $group (keys %{$yaml}) {
+    {
```
Comment (Martchus) [inline lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm]: Keeping the indentation level makes this easier to review. But I'd prefer to get rid of it before the final merge.

Comment (kalikiana) [reply lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm]: Yeah, that was my intention, apparently my comment got lost when editing the description. I wish such a hack wasn't needed... I hate destroying diffs with indentation changes...

File: lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm [hunk 2]
```diff
@@ -236,151 +266,157 @@ sub update {
...
+        # Push the exception to the list of errors without the trailing new line
+        push @$errors, substr($_, 0, -1);
+    };
 
-        # Process only one group
-        last;
+    if (@$errors) {
+        $json->{error} = \@$errors;
+        $self->app->log->error(@$errors);
+        $self->respond_to(json => {json => $json, status => 400},);
```
Comment (kraih) [inline lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm]: That trailing comma doesn't belong there.

Comment (Martchus) [reply lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm]: It is annoying that tidy doesn't remove commas when turning multi-line lists into one liners.

**Pull request merged**